src/project/resources/ResourceDatabase.ts

```diff
@@ -235,26 +235,22 @@ class GDResourceCollection extends ActualResourceCollection<Profile> {
                 }
 
                 const frames: SpriteFrameResource[] = [];
-                const sheetName = removeQualityDecorators(basename(sheetPath));
                 const sheet = new SpriteSheetResource(this.getSource(), sheetPath);
+                const regex = /<key>(?<image>.+?\.png)<\/key>/g;
+                const sheetFile = readFileSync(sheetPath, "utf8");
+                let match;
 
-                // Read sheet data and find all *.png strings inside
-                readFileSync(sheetPath)
-                    .toString()
-                    .match(/\w+\.png/g)
-                    ?.forEach((match) => {

```

Comment:
This solution doesn't provide groups, hence replacing it with a match while loop